server/server.go

```diff
@@ -68,15 +70,87 @@ func main() {
 		fmt.Fprintf(w, "web-api-gateway version: %s\nGo version: %s", version, runtime.Version())
 	})
 
-	logger := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Starting web-api-gateway, version %s\n", version)
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 		log.Printf("Incoming Request %s %s %s", r.RemoteAddr, r.Method, r.URL)
 		http.DefaultServeMux.ServeHTTP(w, r)
 	})
 
-	log.Printf("Starting web-api-gateway, version %s\n", version)
-	log.Fatal(http.ListenAndServeTLS(*addr, *certFile, *keyFile, logger))
+	cr, err := NewCertificateReloader(*certFile, *keyFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	server := &http.Server{
+		Addr: ":https",
+		TLSConfig: &tls.Config{
+			GetCertificate: cr.GetCertificateFunc(),
+		},
+		Handler: mux,
+	}
+	log.Fatal(server.ListenAndServeTLS("", ""))
 }
 
+///////////////////////////////////////////////
+type certificateReloader struct {
+	certMu   sync.RWMutex
+	cert     *tls.Certificate
+	certPath string
+	keyPath  string
+}
+
+func NewCertificateReloader(certPath, keyPath string) (*certificateReloader, error) {
+	cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return nil, err
+	}
+	result := &certificateReloader{
+		certPath: certPath,
+		keyPath:  keyPath,
+	}
+	result.cert = &cert
+
+	// TODO: switch to some other time interval?
+	tickerChannel := time.NewTicker(time.Minute * 30).C
```

Comment:
Would it make sense to provide the interval as a flag, or at least set it as a constant at the top of the file?